### PR TITLE
cyhostboot configuration arguments improvement

### PR DESCRIPTION
**Pull request by peetervois**
+ odd / even / no parity
+ the secure key option

### CODE REVIEW

File: cyhostboot/src/cyhostboot.c [hunk 1]
```diff
@@ -189,8 +247,24 @@ int main(int argc, char **argv)
 	if (action == PROGRAM)
 		printf("Programing file %s\n", args_info.file_arg);
 
+	if (args_info.key_given )
+	{
+		for( int i=0; i<12; i+=2 )
```
Comment (clementleger) [inline cyhostboot/src/cyhostboot.c]: Could you add a define for '12' ?

File: cyhostboot/src/cyhostboot.c [hunk 2]
```diff
@@ -189,8 +247,24 @@ int main(int argc, char **argv)
...
+		{
+			unsigned char k = 0;
+			unsigned char *c = (unsigned char *)&args_info.key_arg[i];
+
+			k|=hex_decode( c[0] )<<4;
+			k|=hex_decode( c[1] );
```
Comment (clementleger) [inline cyhostboot/src/cyhostboot.c]: You should use strtol with a base 16. It will simplify this parsing.

**Review by clementleger — COMMENTED** [review]
Hi Peter, your commits looks good ! However, could you try to keep the existing coding style ?
ie braces on the same line as 'if' and no space after parenthesis.

File: cyhostboot/src/cyhostboot.c (hunk 2 above)
Comment (peetervois) [reply cyhostboot/src/cyhostboot.c]: done.

File: cyhostboot/src/cyhostboot.c (hunk 1 above)
Comment (peetervois) [reply cyhostboot/src/cyhostboot.c]: done.

**Comment by peetervois** [comment]
changed the implementation of key parsing. Now it looks for (comma, semicolon,...) separated string of unsigned numeric values, like 0x12,30,40,017,0x5,6 => 18,30,40,15,5,6

**Comment by clementleger** [comment]
Merged ! Thanks !

**Pull request merged**